vlasiator.cpp

```diff
@@ -227,7 +227,12 @@ void computeNewTimeStep(dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpi
    Real subcycleDt;
 
    // reduce/increase dt if it is too high for any of the three propagators or too low for all propagators
-   if ((P::dt > dtMaxGlobal[0] * P::vlasovSolverMaxCFL ||
+      if (P::dt == P::dt_ceil) {
+          isChanged = false;
```

Comment:
Note: This means that if the timestep ever reaches the ceiling, it can never become smaller again. That's dangerous!

Instead, add  `((P::dt_ceil > 0)&&(P::dt>P::dt_ceil))` to the if-clause-list on line 235-240.

Reply:
ah now I see the continue-mechanism Yann talked about. Not so straightforward, then...

Reply:
Went through the logic again, and in fact the above suggested change *should* do the trick, if I'm correct.